lonboard/geopandas.py

```diff
@@ -0,0 +1,324 @@
+        "CartoDB Positron No Label": basemap.CartoBasemap.PositronNoLabels,
+        "CartoDB Darkmatter": basemap.CartoBasemap.DarkMatter,
+        "CartoDB Darkmatter No Label": basemap.CartoBasemap.DarkMatterNoLabels,
+        "CartoDB Voyager": basemap.CartoBasemap.Voyager,
+        "CartoDB Voyager No Label": basemap.CartoBasemap.VoyagerNoLabels,
+    }
+
+    if map_kwargs is None:
+        map_kwargs = dict()
+    if classify_kwargs is None:
+        classify_kwargs = dict()
+    if layer_kwargs is None:
+        layer_kwargs = dict()
+    if isinstance(elevation, str):
+        if elevation in gdf.columns:
+            elevation = gdf[elevation]
+        else:
+            raise ValueError(
+                f"the designated height column {elevation} is not in the dataframe"
+            )
+        if not pd.api.types.is_numeric_dtype(elevation):
+            raise ValueError("elevation must be a numeric data type")
+
+    if not pd.api.types.is_list_like(nan_color):
+        raise ValueError("nan_color must be an iterable of 3 or 4 values")
+
+    if len(nan_color) != 4:
+        if len(nan_color) == 3:
+            nan_color = np.append(nan_color, [255])
+        else:
+            raise ValueError("nan_color must be an iterable of 3 or 4 values")
+
+    # only polygons have z
+    if ["Polygon", "MultiPolygon"] in gdf.geometry.geom_type.unique():
+        layer_kwargs["get_elevation"] = elevation
+        layer_kwargs["extruded"] = extruded
+        layer_kwargs["elevation_scale"] = elevation_scale
+        layer_kwargs["wireframe"] = wireframe
+
+    LINE = False  # set color of lines, not fill_color
+    if ["LineString", "MultiLineString"] in gdf.geometry.geom_type.unique():
+        LINE = True
+    if color:
+        if LINE:
+            layer_kwargs["get_color"] = color
+        else:
+            layer_kwargs["get_fill_color"] = color
+    if column is not None:
+        if column not in gdf.columns:
+            raise ValueError(f"the designated column {column} is not in the dataframe")
+        if gdf[column].dtype in ["O", "category"]:
+            categorical = True
+        if cmap is not None and cmap not in colormaps:
+            raise ValueError(
+                f"`cmap` must be one of {list(colormaps.keys())} but {cmap} was passed"
+            )
+        if cmap is None:
+            cmap = "tab20" if categorical else "viridis"
+        if categorical:
+            color_array = _get_categorical_cmap(gdf[column], cmap, nan_color, alpha)
+        elif scheme is None:
+            # minmax scale the column first, matplotlib needs 0-1
+            transformed = (gdf[column] - np.nanmin(gdf[column])) / (
+                np.nanmax(gdf[column]) - np.nanmin(gdf[column])
+            )
+            color_array = apply_continuous_cmap(
+                values=transformed, cmap=colormaps[cmap], alpha=alpha
+            )
+        else:
+            color_array = get_color_array(
```

Comment:
This is the only place where `get_color_array` is used, right? We should only import it within this block.